spec/services/ccms/case_add_requestor_xml_blocks_spec.rb

```diff
@@ -519,6 +499,25 @@ module CCMS # rubocop:disable Metrics/ModuleLength
         end
       end
 
+      context 'EMERGENCY_FURTHER_INFORMATION' do
+        context 'delegated function used' do
+          it 'returns hard coded statement' do
+            allow(legal_aid_application).to receive(:used_delegated_functions?).and_return(true)
+            allow(legal_aid_application).to receive(:used_delegated_functions_on).and_return(Date.today)
```

Comment:
`legal_aid_application` is defined via a factory and `used_delegated_functions` and `used_delegated_functions_on` are both attributes of the model.

So override the factory rather than using mocking.

Also `block` is the same for both `it` declarations
```
    context 'delegated function used' do
      let(:block) { XmlExtractor.call(xml, :global_merits, 'EMERGENCY_FURTHER_INFORMATION') } 
      let(:legal_aid_application) do
        create :legal_aid_application,
               :with_everything,
               :with_applicant_and_address,
               populate_vehicle: true,
               with_bank_accounts: 2,
               proceeding_types: [proceeding_type],
               provider: provider,
               used_delegated_functions: true,
               used_delegated_functions_on: Date.today
      end

      it 'returns hard coded statement' do
        expect(block).to have_text_response 'Apply Service application - see uploaded provider statement of case'
      end
      ....
```

Reply:
Thanks. Good points. I didn't actually write those specs, I just moved them around a little to make them clearer (there was a problem with one `context` being nested inside an unrelated `context`) but your suggestions make sense and will bring the tests in line with the others in the file, so I've fixed them.